plugins/branch_name_in_commit/__init__.py

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python2
+
+def build_filter(args):
+    return Filter(args)
+
+class Filter:
+    def __init__(self, args):
+        if not args in ['start','end']:
+            raise Exception('Cannot have branch name anywhere but start and end')
+        self.pos = args
+
+    def commit_message_filter(self,hash):
```

Comment:
`hash` is the type of the argument, call it `commit_data` or something more descriptive. (this goes for all the filters)